build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+                        } else {
+                            echo "PR build definition *not* detected: building with 'make jenkins'"
+                            sh "make jenkins CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                        }
+                    }
+                }
+
+                echo "Stage result: build: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage('create vsix') {
+            options {
+                timeout(time: 30, unit: 'MINUTES')  // Typically takes less than 5 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    sh "make create-vsix CONFIGURATION=${env.BuildFlavor} V=1"
+                }
+
+                echo "Stage result: create vsix: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage('build tests') {
+            options {
+                timeout(time: 30, unit: 'MINUTES')       // Typically takes 10 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    sh "make all-tests CONFIGURATION=${env.BuildFlavor} V=1"
+                }
+
+                echo "Stage result: build tests: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage ('process build results') {
+            options {
+                timeout(time: 10, unit: 'MINUTES')      // Typically takes less than a minute
+            }
+
+            steps {
+                script {
+                    dir("$XADir") {
+                        def status = processBuildStatus()
+
+                        if (status == 0) {
+                            buildStatusProcessed = true
+                        }
+                    }
+                }
+
+                echo "Stage result: process build results: ${currentBuild.currentResult}"
+            }
+        }
+
+        stage('publish packages to Azure') {
+            options {
+                timeout(time: 10, unit: 'MINUTES')    // Typically takes less than 2 minutes
+            }
+
+            steps {
+                script {
+                    def status = publishPackages(publishBuildFilePaths)
+                    if (status == 0) {
+                        buildStatusUploaded = true
+                    }
+                }
+
+                echo "Stage result: publish packages to Azure: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage('run all tests') {
+            options {
+                timeout(time: 160, unit: 'MINUTES')     // XA tests typically take 1 hr, 40 minutes to run; timeout after 2 hrs, 40 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    script {
+                        def status = 0
+                        try
+                        {
```

Comment:
Why is this `{` on a newline?

Reply:
Fixed this rogue brace with commit https://github.com/xamarin/xamarin-android/commit/c570faf9b695b5bb7059d9fa8276a788c8b83169